src/main/java/org/openstreetmap/atlas/utilities/testing/OsmosisXmlReaderFromResource.java

```diff
@@ -0,0 +1,162 @@
+package org.openstreetmap.atlas.utilities.testing;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.osmosis.core.OsmosisRuntimeException;
+import org.openstreetmap.osmosis.core.task.v0_6.RunnableSource;
+import org.openstreetmap.osmosis.core.task.v0_6.Sink;
+import org.openstreetmap.osmosis.xml.common.CompressionActivator;
+import org.openstreetmap.osmosis.xml.common.CompressionMethod;
+import org.openstreetmap.osmosis.xml.v0_6.XmlReader;
+import org.openstreetmap.osmosis.xml.v0_6.impl.OsmHandler;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+/**
+ * From {@link XmlReader}. Almost everything here is copied from osmosis-xml version 0.44.1. The
+ * only change is to allow reading from a {@link Resource} instead from a {@link File}.
+ *
+ * @author Brett Henderson
+ * @author matthieun
+ */
+public class OsmosisXmlReaderFromResource implements RunnableSource
+{
+    private static Logger log = Logger.getLogger(XmlReader.class.getName());
```

Comment:
Nit: Update logger name to be OsmosisXmlReaderFromResource

Reply:
Nice catch!